### PR TITLE
Box alphabetical imports linter.

### LINKED ISSUE #543: Box import alphabetical
Require packages and modules sorted alphabetically.

Require functions sorted alphabetically.

### PR DESCRIPTION
**Pull request by radbasa**
### Changes
- Closes #543
- Adds `box_alphabetical_imports_linter()`
- Adds the linter to the `.lintr` template
- Adds documentation
- This linter ignores package/module/function aliases.

### How to test
1. Unit tests for `box_alphabetical_imports_linter()` is provided.

### CODE REVIEW

File: R/box_linters.R [hunk 1]
```diff
@@ -1,3 +1,124 @@
...
+  xpath_modules_with_functions <- paste(xpath_base, "
+/child::expr[
+  descendant::SYMBOL and
+  descendant::OP-LEFT-BRACKET
+]")
+
+  xpath_functions <- "./descendant::expr/SYMBOL[
+  ../preceding-sibling::OP-LEFT-BRACKET and
+  ../following-sibling::OP-RIGHT-BRACKET
+]"
```
Comment (jakubnowicki) [inline R/box_linters.R:80]: Please add indentation.

File: R/box_linters.R [hunk 2]
```diff
@@ -1,3 +1,124 @@
+# nolint start: line_length_linter
+#' `box` library alphabetical module and function imports linter
+#'
+#' Checks that module and function imports are sorted alphabetically.
+#' See the [Explanation: Rhino style guide](https://appsilon.github.io/rhino/articles/explanation/rhino-style-guide.html)
+#' to learn about the details.
+#'
+#' @return A custom linter function for use with `r-lib/lintr`.
+#'
+#' @examples
```
Comment (jakubnowicki) [inline R/box_linters.R:10]: Please add examples with aliases.

File: R/box_linters.R [hunk 3]
```diff
@@ -1,3 +1,124 @@
+# nolint start: line_length_linter
+#' `box` library alphabetical module and function imports linter
+#'
+#' Checks that module and function imports are sorted alphabetically.
+#' See the [Explanation: Rhino style guide](https://appsilon.github.io/rhino/articles/explanation/rhino-style-guide.html)
```
Comment (jakubnowicki) [inline R/box_linters.R:5]: Please update the style guide with an aliases case.

File: tests/testthat/test-linter_box_alphabetical.R
Comment (jakubnowicki) [inline tests/testthat/test-linter_box_alphabetical.R:1]: Please add a test case with aliases.

**Review by jakubnowicki — COMMENTED** [review]
Works fine. I'm ok with how it handles aliases, but please make it clear in the documentation, what is the approach.

File: tests/testthat/test-linter_box_alphabetical.R [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    shiny,
+    tidyr[long, pivot, wide],
+  )"
+
+  good_box_calls_alias_4 <- "box::use(
+    path/to/fileA[functionA, alias = functionB, functionC],
+    path/to/fileB,
+    path/to/fileC[functionD, functionE, functionF],
+  )"
+
```
Comment (radbasa) [inline tests/testthat/test-linter_box_alphabetical.R:60]: tests for alias were already there.

**Review by jakubnowicki — APPROVED** [review]
LGTM

**Pull request merged**